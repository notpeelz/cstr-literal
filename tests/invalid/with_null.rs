use cstr_literal::cstr;

fn main() {
  cstr!("foo\0bar");
}
