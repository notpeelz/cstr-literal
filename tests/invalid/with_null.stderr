error[E0080]: evaluation of constant value failed
 --> src/lib.rs
  |
  |         [0, _, ..] => panic!("C strings can't contain null bytes"),
  |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the evaluated program panicked at 'C strings can't contain null bytes', $DIR/src/lib.rs:15:23
  |
note: inside `new`
 --> src/lib.rs
  |
  |         [0, _, ..] => panic!("C strings can't contain null bytes"),
  |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: inside `__CSTR`
 --> tests/invalid/with_null.rs:4:3
  |
4 |   cstr!("foo\0bar");
  |   ^^^^^^^^^^^^^^^^^
  = note: this error originates in the macro `$crate::panic::panic_2021` which comes from the expansion of the macro `cstr` (in Nightly builds, run with -Z macro-backtrace for more info)
